rotkehlchen/chain/ethereum/modules/eth2.py

```diff
@@ -23,9 +56,321 @@ def __init__(
+        depositing_indices = set(index_to_address.keys()) - set(performance_result.keys())
+        for index in depositing_indices:
+            stats = self.get_validator_daily_stats(
+                validator_index=index,
+                msg_aggregator=beaconchain.msg_aggregator,
+            )
+            result.append(ValidatorDetails(
+                validator_index=index,
+                public_key=index_to_pubkey[index],
+                eth1_depositor=index_to_address[index],
+                performance=DEPOSITING_VALIDATOR_PERFORMANCE,
+                daily_stats=stats,
+            ))
+
+        return result
+
+    def get_validator_daily_stats(
+            self,
+            validator_index: int,
+            msg_aggregator: MessagesAggregator,
+            from_timestamp: Optional[Timestamp] = None,
+            to_timestamp: Optional[Timestamp] = None,
+    ) -> List[ValidatorDailyStats]:
+        """Gets the daily stats of an ETH2 validator by index
+
+        First queries the DB for the already known stats and then if needed also scrapes
+        the beacocha.in website for more. Saves all new entries to the DB.
+        """
+        dbeth2 = DBEth2(self.database)
+        known_stats = dbeth2.get_validator_daily_stats(
+            validator_index=validator_index,
+            from_ts=from_timestamp,
+            to_ts=to_timestamp,
+        )
+        last_ts = Timestamp(0) if len(known_stats) == 0 else known_stats[-1].timestamp
+        limit_ts = to_timestamp if to_timestamp else ts_now()
+        if limit_ts - last_ts <= DAY_IN_SECONDS:
+            return known_stats  # no need to requery if less than a day passed
+
+        now = ts_now()
+        should_backoff = (
+            now - self.last_stats_query_ts < VALIDATOR_STATS_QUERY_BACKOFF_TIME_RANGE and
+            self.validator_stats_queried >= VALIDATOR_STATS_QUERY_BACKOFF_EVERY_N_VALIDATORS
+        )
+        if should_backoff:
+            log.debug(
+                f'Queried {self.validator_stats_queried} validators in the last '
+                f'{VALIDATOR_STATS_QUERY_BACKOFF_TIME_RANGE} seconds. Backing off for '
+                f'{VALIDATOR_STATS_QUERY_BACKOFF_TIME} seconds.',
+            )
+            self.validator_stats_queried = 0
+            gevent.sleep(VALIDATOR_STATS_QUERY_BACKOFF_TIME)
+
+        new_stats = scrape_validator_daily_stats(
+            validator_index=validator_index,
+            last_known_timestamp=last_ts,
+            msg_aggregator=msg_aggregator,
+        )
+        self.validator_stats_queried += 1
+        self.last_stats_query_ts = now
+
+        if len(new_stats) != 0:
+            dbeth2.add_validator_daily_stats(stats=new_stats)
+
+        return dbeth2.get_validator_daily_stats(
+            validator_index=validator_index,
+            from_ts=from_timestamp,
+            to_ts=to_timestamp,
+        )
 
+    # -- Methods following the EthereumModule interface -- #
```

Comment:
At some point we should make use or get rid of those methods

Reply:
We need those for all modules. All modules should use them. Just don't wanna add too many changes in the PR.

For example eth2 should query eth2 balances for a new address from here. We already do it in the balance query, but it should go here.